frontend/src/bundles/auth/store/slice.ts

```diff
@@ -2,22 +2,39 @@ import { createSlice } from '@reduxjs/toolkit';
 
 import { DataStatus } from '~/bundles/common/enums/enums.js';
 import { type ValueOf } from '~/bundles/common/types/types.js';
+import { type UserSignInResponseDto } from '~/bundles/users/users.js';
 
-import { signUp } from './actions.js';
+import { signIn, signUp } from './actions.js';
 
 type State = {
     dataStatus: ValueOf<typeof DataStatus>;
+    user: UserSignInResponseDto | undefined;
+    errorMessage: string | undefined;
 };
 
 const initialState: State = {
     dataStatus: DataStatus.IDLE,
+    user: undefined,
+    errorMessage: undefined,
```

Comment:
replace undefined with null

Reply:
I use `undefined` for errorMessage because on rejected case I assign `state.errorMessage = action.error.message` which has `string | undefined` type. And for user I use undefined for consistency.

Reply:
Error messages from the backend we will display as Toast notification so there is no need to have this field in store

You will get the error messages that we need from zod

Reply:
What about this type of error (when user with provided email does not exist or password is invalid)?
<img width="694" alt="Screenshot 2024-08-19 at 16 57 38" src="https://github.com/user-attachments/assets/2b15b240-4b6c-4d8e-90aa-3937db05f285">

Reply:
Skip it for now, we'll also handle it in our middleware